### PR TITLE
Add CPOs for for_loop algorithms

### PR DESCRIPTION
**Pull request by msimberg**
Can go in 1.5.0 or 1.6.0. This should be fairly straightforward (i.e. no big surprises with builds and tests), but let's see.

### CODE REVIEW

**Comment by gonidelis** [comment]
I like the doc fixes. @hkaiser Shouldn't the doxygen comments regarding the no-exec-policy overloads be added in the other algorithms that have already been adapted, too? Plus, should we link this to #4822?

**Comment by msimberg** [comment]
@gonidelis does this address your comment about other algorithms needing updating: https://github.com/STEllAR-GROUP/hpx/pull/4908?

**Comment by gonidelis** [comment]
@msimberg  Yeah. That's great! Plus, we need to fix these doxygen comments that concern the overloads that do not take an exec-policy argument like you did on this PR.

File: docs/sphinx/api/public_api.rst [hunk 1]
```diff
@@ -92,10 +92,10 @@ Functions
...
 - :cpp:func:`hpx::parallel::v1::unique`
 - :cpp:func:`hpx::parallel::v1::unique_copy`
-- :cpp:func:`hpx::parallel::v2::for_loop`
-- :cpp:func:`hpx::parallel::v2::for_loop_strided`
-- :cpp:func:`hpx::parallel::v2::for_loop_n`
-- :cpp:func:`hpx::parallel::v2::for_loop_n_strided`
+- :cpp:func:`hpx::for_loop`
+- :cpp:func:`hpx::for_loop_strided`
+- :cpp:func:`hpx::for_loop_n`
+- :cpp:func:`hpx::for_loop_n_strided`
```
Comment (hkaiser) [inline docs/sphinx/api/public_api.rst:98]: Should these go into `hpx::experimental` instead (it's not in the standard yet, after all)

File: docs/sphinx/manual/writing_single_node_hpx_applications.rst [hunk 2]
```diff
@@ -749,23 +749,23 @@ Parallel algorithms
...
      * Implements loop functionality over a range specified by integral or iterator bounds.
-     * ``<hpx/include/parallel_for_loop.hpp>``
-   * * :cpp:func:`hpx::parallel::v2::for_loop_n`
+     * ``<hpx/algorithm.hpp>``
+   * * :cpp:func:`hpx::for_loop_n`
      * Implements loop functionality over a range specified by integral or iterator bounds.
-     * ``<hpx/include/parallel_for_loop.hpp>``
-   * * :cpp:func:`hpx::parallel::v2::for_loop_n_strided`
+     * ``<hpx/algorithm.hpp>``
+   * * :cpp:func:`hpx::for_loop_n_strided`
```
Comment (hkaiser) [inline docs/sphinx/manual/writing_single_node_hpx_applications.rst:766]: Thanks for changing it here as well... I think we have missed correcting all the other algorithms we have already ported to use CPOs here...

File: examples/quickstart/disable_thread_stealing_executor.cpp [hunk 3]
```diff
@@ -120,10 +120,10 @@ int main()
 
     // The following for_loop will be executed while thread stealing is disabled
     auto exec = executor_example::make_disable_thread_stealing_executor(
-        hpx::parallel::execution::par.executor());
+        hpx::execution::par.executor());
 
-    hpx::parallel::for_loop(hpx::parallel::execution::par.on(exec), 0, v.size(),
-        [](std::size_t i) {});
+    hpx::for_loop(
+        hpx::execution::par.on(exec), 0, v.size(), [](std::size_t i) {});
```
Comment (hkaiser) [inline examples/quickstart/disable_thread_stealing_executor.cpp:126]: Not to self: this change should be done in a separate PR for all occurrences.

File: docs/sphinx/manual/writing_single_node_hpx_applications.rst (hunk 2 above)
Comment (hkaiser) [reply docs/sphinx/manual/writing_single_node_hpx_applications.rst:766]: See #4908.

**Comment by hkaiser** [comment]
Do you plan to still add range-based versions here? We can add those in a separate PR as well.

File: docs/sphinx/api/public_api.rst (hunk 1 above)
Comment (msimberg) [reply docs/sphinx/api/public_api.rst:98]: Unsure. I'd be inclined to leave it here because they're stable (enough) in HPX, even though they are experimental in the standard. Our executors are also not in the standard, but not in the experimental namespace.

File: docs/sphinx/manual/writing_single_node_hpx_applications.rst (hunk 2 above)
Comment (msimberg) [reply docs/sphinx/manual/writing_single_node_hpx_applications.rst:766]: I know you already saw the PR, but for anyone else wondering about this: #4908.

File: examples/quickstart/disable_thread_stealing_executor.cpp (hunk 3 above)
Comment (msimberg) [reply examples/quickstart/disable_thread_stealing_executor.cpp:126]: Yes, I only changed the ones close to the lines I changed. We'll eventually have to deprecate the names in `hpx::parallel::execution` which will help us change everything to the new names.

**Comment by msimberg** [comment]
> Do you plan to still add range-based versions here? We can add those in a separate PR as well.

I was not planning on doing that here, but can definitely do it in a separate PR. I suppose there's no document for those, but they would just replace `first` and `last` with `range`?

**Comment by hkaiser** [comment]
> I suppose there's no document for those, but they would just replace `first` and `last` with `range`?

Yes, that's what we've done for the other algorithms before C++20. This also might entail adding sentinel support to the machinery. But even that can be done separately.

**Comment by msimberg** [comment]
@hkaiser anything against this still going in 1.5.0?

**Comment by hkaiser** [comment]
> @hkaiser anything against this still going in 1.5.0?

No objections here. Please go ahead.

**Review by aurianer — APPROVED** [review]
LGTM, thanks!
